core/src/main/java/org/elasticsearch/ingest/ConfigurationUtils.java

```diff
@@ -256,6 +257,11 @@ private static Processor readProcessor(ProcessorsRegistry processorRegistry, Str
             List<Processor> onFailureProcessors = readProcessorConfigs(onFailureProcessorConfigs, processorRegistry);
             Processor processor;
             processor = factory.create(config);
+
+            if (onFailureProcessorConfigs != null && onFailureProcessors.isEmpty() == true) {
```

Comment:
instead of `onFailureProcessors.isEmpty() == true`, just `onFailureProcessors.isEmpty()`?